### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by sigbot**
Add CI builds with Azure Pipelines

### CODE REVIEW

**Comment by HotelCalifornia** [comment]
it build!

note that there's no reason to build macOS for now given that we use homebrew (which just builds on the target machine). in the future we may want to look at building the whl files for mac (unless they're platform independent? can't remember), but until it becomes an issue I think this is good to go.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      x64:
+        buildArch: x64
+      x86:
+        buildArch: x86
+  pool:
+    vmImage: ubuntu-16.04
+  steps:
+    - bash: sudo apt-add-repository ppa:deadsnakes/ppa
+    - bash: sudo apt-get update
+    - bash: sudo apt-get install python3.6 python3.6-dev python3-setuptools
```
Comment (HotelCalifornia) [inline azure-pipelines.yml:60]: I know that we usually prefer to have this kind of stuff in a script file, but in this case it's 1) [hopefully] temporary and 2) very specifically only an issue on azure linux agents

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#     - task: UsePythonVersion@0
```
Comment (edjubuh) [inline azure-pipelines.yml:62]: Do you mind adding a comment referencing the GH issue why this won't work?

Microsoft/azure-pipelines-image-generation#225

**Review by edjubuh — CHANGES_REQUESTED** [review]
Looks good pls add comment

File: azure-pipelines.yml (hunk 1 above)
Comment (edjubuh) [reply azure-pipelines.yml:60]: Agreed

File: azure-pipelines.yml (hunk 2 above)
Comment (HotelCalifornia) [reply azure-pipelines.yml:62]: added above

**Review by edjubuh — APPROVED** [review]
Make it so

**Comment by HotelCalifornia** [comment]
aw hell. @sigbot is ruining things because of its non-signed commit. gonna have to invoke admin abuse :-/

**Pull request merged**